src/globals.ts

```diff
@@ -33,12 +42,18 @@ export class SpansContainer {
       this.totalSpans += 1;
     }
     // Memory optimization, take up to 10x maxSize because of smart span selection logic
-    if (spanHasErrors(span) || getMaxRequestSizeOnError() * 10 > this.currentSpansSize) {
+    const maxSpansSize = getMaxSizeForStoredSpansInMemory();
+    if (spanHasErrors(span) || maxSpansSize > this.currentSpansSize) {
```

Comment:
I think it's a bit clearer to reverse the condition, and also make this `<=` (in other place, we truncate the payload if it's greater than the limit, meaning a value equal to the limit is allowed - so I think we better be consistent)
```suggestion
    if (spanHasErrors(span) || this.currentSpansSize <= maxSpansSize) {
```